cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java

```diff
@@ -1684,6 +1739,77 @@ TSStatus forwardPlan(List<PartitionGroup> partitionGroups, PhysicalPlan plan) {
     return status;
   }
 
+  /**
+   * Create timeseries automatically
+   *
+   * @param insertPlan,    some of the timeseries in it are not created yet
+   * @return true of all uncreated timeseries are created
+   */
+  boolean autoCreateTimeseries(InsertPlan insertPlan) {
+    List<String> seriesList = new ArrayList<>();
+    String deviceId = insertPlan.getDeviceId();
+    String storageGroupName;
+    try {
+      storageGroupName = MetaUtils
+          .getStorageGroupNameByLevel(deviceId, IoTDBDescriptor.getInstance()
+              .getConfig().getDefaultStorageGroupLevel());
+    } catch (MetadataException e) {
+      logger.error("Failed to infer storage group from deviceId {}", deviceId);
+      return false;
+    }
+    for (String measurementId : insertPlan.getMeasurements()) {
+      seriesList.add(
+          new StringContainer(new String[]{deviceId, measurementId}, TsFileConstant.PATH_SEPARATOR)
+              .toString());
+    }
+    PartitionGroup partitionGroup = partitionTable.route(storageGroupName, 0);
+    List<String> unregisteredSeriesList = getUnregisteredSeriesList(seriesList, partitionGroup);
+    for (String seriesPath : unregisteredSeriesList) {
+      int index = seriesList.indexOf(seriesPath);
+      TSDataType dataType = TypeInferenceUtils
+          .getPredictedDataType(insertPlan.getValues()[index], true);
+      TSEncoding encoding = getDefaultEncoding(dataType);
+      CompressionType compressionType = TSFileDescriptor.getInstance().getConfig().getCompressor();
+      CreateTimeSeriesPlan createTimeSeriesPlan = new CreateTimeSeriesPlan(new Path(seriesPath),
+          dataType, encoding, compressionType, null, null, null, null);
+      // TODO-Cluster: add executeNonQueryBatch()
+      TSStatus result = executeNonQuery(createTimeSeriesPlan);
```

Comment:
maybe you can use `processPartitionedPlan` directly to avoid unnecessary judgement

Reply:
You are right. Thanks.